packages/graphback-runtime-mongodb/tests/MongoDataProviderTest.ts

```diff
@@ -1,154 +1,369 @@
+    }
+    
+  });
+
+  it('can filter using OR', async () => {
+    const posts: Post[] = await context.providers.Post.findBy({
+      or: [
+        {
+          text: { eq: 'post2' }
+        },
+        {
+          likes: { eq: 300 }
+        }
+      ]
     });
-  }
-  const todos = await context.provider.findBy({ text }, { limit: 1 });
-  expect(todos[0].text).toEqual(text);
-});
+    expect(posts.length).toEqual(2);
+    for(const post of posts) {
+      expect((post.text === 'post2') || (post.likes === 300));
+    }
+  });
 
-test('find first 1 todos by text', async () => {
-  const text = 'todo-test';
-  for (let i = 0; i < 2; i++) {
-    await context.provider.create({
-      text,
+  it('can filter using NOT', async () => {
+    const posts: Post[] = await context.providers.Post.findBy({
+      not: [
+        {
+          text: { eq: 'post2' }
+        },
+        {
+          likes: { eq: 300 }
+        }
+      ]
     });
-  }
 
-  const todos = await context.provider.findBy({ text } , { limit: 1, offset: 0});
-  expect(todos.length).toEqual(1);
-  expect(todos[0].text).toEqual(text);
-});
+    expect(posts.length).toBeGreaterThanOrEqual(1);
+    for(const post of posts) {
+      expect(post.text).not.toEqual('post2');
+      expect(post.likes).not.toEqual(300);
+    }
+  });
+
+  it('can filter using between operator', async () => {
+    const posts :Post[] = await context.providers.Post.findBy({
+          likes: { between: [250, 350] }
+    });
+
+    expect(posts.length).toBeGreaterThanOrEqual(1);
+    for (const post of posts) {
+      expect(post.likes).toBeLessThanOrEqual(350);
+      expect(post.likes).toBeGreaterThanOrEqual(250);
+    }
+  });
+
+  it('can filter using nbetween operator', async () => {
+    const posts :Post[] = await context.providers.Post.findBy({
+      likes: { nbetween: [250, 350] }
+    });
+
+    expect(posts.length).toBeGreaterThanOrEqual(1);
+    for (const post of posts) {
+      expect((post.likes < 250) || (post.likes > 350));
+    }
+  });
+  Object.defineProperty(RegExp.prototype, "toJSON", {
+    value: RegExp.prototype.toString
+  });
+  it('can use nested filters', async () => {
+    const posts :Post[] = await context.providers.Post.findBy({
+      and: [
+        {
+          or: [
+            { likes: { between: [250, 350] } },
+            { likes: { between: [25, 75] } }
+          ]
+        },
+        {
+          or: [
+            { text: { eq: 'post' } },
+            { text: { eq: 'post2' } }
+          ]
+        }
+      ]
+    });
+
+    console.log(JSON.stringify(posts,null,4))
+    expect(posts.length).toBeGreaterThanOrEqual(1);
+    for (const post of posts) {
+      expect(
+        (
+          (post.likes >= 250) && (post.likes <= 350)
+        )
+        ||
+        (
+          (post.likes >= 25) && (post.likes <= 75)
+        )
+      );
+
+      expect(
+        (post.text === 'post')
+        ||
+        (post.text === 'post2')
+      );
```

Comment:
Is there some alternative to using matchers like this? I looked but I couldn't find anything